packages/valory/skills/market_manager_abci/bets.py

```diff
@@ -262,6 +313,14 @@ def hook(data: Dict[str, Any]) -> Union[Bet, PredictionResponse, Dict[str, Bet]]
         bet_annotations = sorted(Bet.__annotations__.keys())
         if bet_annotations == data_attributes:
             return Bet(**data)
+        else:
+            # fetch missing attributes from the data
+            missing_attributes = set(bet_annotations) - set(data_attributes)
+            new_attributes = {"queue_status", "invested_amount"}
+            if missing_attributes == new_attributes:
+                data["queue_status"] = QueueStatus(0)
+                data["invested_amount"] = 0
```

Comment:
I wonder if this is necessary since there are already default values in the dataclass definition.

If I am not missing something, then we could reformat this code to something like:
1. Create constant for new attributes.
2. Calc the sets and the missing attributes.
3. if `bet_annotations` are the same as the `data_attributes` or `missing_attributes` are the `new_attributes` then `return Bet(**data)`

Reply:
Also, we need to make sure that the users can also roll back to previous releases with the edited `bets.json` files. Therefore we should also add a check on whether the new attributes are present in the file but not in the dataclass and then pop them before decoding.